src/odemis/gui/util/widgets.py

```diff
@@ -354,3 +355,41 @@ def _update_progress(self):
             # wxST_ELLIPSIZE_END in xrc) or dc.GetTextExtend()
             self._label.SetLabel(lbl_txt)
             self._label.Parent.Layout()
+
+
+class EllipsisAnimator(object):
```

Comment:
Making it a child of RepeatingTimer would probably simplify a bit. You could get rid of stop(), and cancel() would be used directly. Especially, the join() is likely not really useful (as it should immediately return).